htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxResponse.java

```diff
@@ -361,16 +330,16 @@ public String getRetarget() {
         return retarget;
     }
 
-    public Map<String, String> getTriggers() {
-        return Collections.unmodifiableMap(this.triggers);
+    public Collection<HtmxTrigger> getTriggers() {
+        return Collections.unmodifiableCollection(this.triggers);
     }
 
-    public Map<String, String> getTriggersAfterSettle() {
-        return Collections.unmodifiableMap(this.triggersAfterSettle);
+    public Collection<HtmxTrigger> getTriggersAfterSettle() {
+        return Collections.unmodifiableCollection(this.triggersAfterSettle);
     }
 
-    public Map<String, String> getTriggersAfterSwap() {
-        return Collections.unmodifiableMap(this.triggersAfterSwap);
+    public Collection<HtmxTrigger> getTriggersAfterSwap() {
+        return Collections.unmodifiableCollection(this.triggersAfterSwap);
     }
```

Comment:
Not sure how to handle it, without breaking the API. I didn't want to introduce different methods for the new return type. IMO we should accept it and explicitly mention this in the release notes. The question is how many users actually use these methods and whether it is worth the effort to be backward compatible.

Reply:
If we do this, we need to go for 4.0.0. I hate it myself if minor updates break backwards compatibility. I would propose to do the other changes in a 3.1.0 and keep this PR open for a 4.0.0 later.

Reply:
Ok, then I would go for package private methods that return `Set<HtmxTrigger>` to use internally and leave the API untouched until 4.0 so things don't get too complicated. In 4.0 we can drop these methods and change the return type of the current methods. But that will all have to wait because I'm on holiday for 2 weeks.

Reply:
Good idea. I changed the assigned milestone back to 3.1.0 on this PR.